component/azstorage/utils.go

```diff
@@ -309,6 +309,27 @@ func parseMetadata(attr *internal.ObjAttr, metadata map[string]*string) {
 	}
 }
 
+func AddMetadata(metadata map[string]*string, key, value string) bool {
+	lowerKey := strings.ToLower(key)
+	var existingKey string
+	var found bool
+	for k := range metadata {
+		if strings.ToLower(k) == lowerKey {
+			existingKey, found = k, true
+			break
+		}
+	}
+	if found {
+		if *metadata[existingKey] == value {
+			return false
+		}
+		metadata[existingKey] = &value
```

Comment:
we modify the value at two places we shall unify this saying if the key was a match and value is same then skip otherwise update the value